### DISCUSSION TITLE
Overview | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#558] [Q&A]
# Overview | Seeed Studio Wiki

Overview

https://wiki.seeedstudio.com/Wio-Terminal-Audio-Overview/

**Comment by simonetravaglini** [comment]
It's not clear how to connect the sound card with WIO, can you explain better? it's possible use grove connector?

**Reply by xzxcessarr** [reply]
Hello! This hat is a connector like the pin, but as we all known that you should connect them which are matching the feature of the pin, so you connect them with the orientation we provide and just follow our wiki: https://wiki.seeedstudio.com/Wio-Terminal-Audio-Overview/#hardware-connection

**Comment by Anja6969** [comment]
how many watts is the build in amp and what should the impedance of the speaker be?

**Reply by Darkextratoasty** [reply]
I know this is very old, but in case others come here wondering the same thing: 
The WM8960 chip supports 1W into an 8ohm speaker in speaker mode and up to 40mW into 16ohm headphones in headphone mode.

**Comment by mirkokral** [comment]
Seems like the SD card can support even more GB on circuitpython (hadn't tested on others), because i used my 128 gigabyte SD card there.

**Reply by mirkokral** [reply]
The SD card size support depends on the used library. Quite stupid of me not to immediately assume that.

**Comment by palacita135** [comment]
Can we attach external mic for this module from the audio jack?

**Comment by zol123** [comment]
So if  You need  use  extrenal   IR  Emiter  with WIO Terminal ( f.e.  battery chassis  cover  IR emiter) 
via  Seeed_Arduino_IR repositorie   edit  file  IRLibSAMD51.h 
line: 
  #define IR_SEND_PWM_PIN WIO_IR    

to:

  #define IR_SEND_PWM_PIN D0 // WIO_IR     (for D0) pin

**Comment by Brandtz1** [comment]
Hi, I still get the Compilation error: Seeed_Arduino_LCD.h: No such file or directory. Despite downloading the ZIP library from Git and installing it...

**Comment by FyrbyAdditive** [comment]
The python script on this page does not work in macOS.

**Comment by y-lobau** [comment]
Does anyone know how does it work with picovoice? Is there an API to customize picovoice? Is it free to use forever or the same pricing requirements and limits apply (https://picovoice.ai/pricing/)?

**Comment by ghost** [comment]
Useless POS.  Never buying a product from this company again.

**Comment by RaphTronic** [comment]
How can files created by Seeed_Arduino_FS show the correct date / time ?

Using the ReadWrite example from the Seeed_Arduino_FS library, all the files generated show a Date Created/Modified value of 12/31/2017 11pm under Windows. Regardless of the WIO RTC's actual date/time at the time of the file creation.

I could not find any WIO Terminal code example that properly initializes the file system's date. On non-WIO Arduino platforms the call  SdFile::dateTimeCallback(dateTime); is used for that purpose but it does not seem to be supported by Seeed_Arduino_FS

**Comment by IrajRehman** [comment]
Hey there, I am using the ReSpeaker 2 mic with Raspberry pi 4B after the driver cloned from the given GitHub link after that there is an error is coming while installing the Install.sh it give [WARNING Your loaded Kernel version is 6.1.21 Not matching the updated version. Kernel was updated , but new kernel was not loaded yet please reboot your machine AND THEN run this script again] after this I rebooted it and again updating the system but when I am trying to again installing but it give this and ReSpeaker also not showing on the raspberry pi mount devices list. Could Someone help me I'll appreciate it.

**Reply by HM-sur** [reply]
What is your current kernel version? 6.1.21 is not supported.

**Reply by IrajRehman** [reply]
Okay, then which version of OS should I use.

**Reply by HM-sur** [reply]
6.6.x or above is preferred

**Comment by ryanfiller** [comment]
Is documentation about how to hook this up to Ollama? The banner image on seeedstudio.com/watcher shows that integration but I can't find any references to it in the github project.

**Reply by Zeke5579** [reply]
You can refer to this document to deploy ollama: https://wiki.seeedstudio.com/watcher_local_deploy/

**Comment by Tygopit** [comment]
Any chance to have the option to save the image (taken upon detection) to the inserted sd card?

**Reply by nfs0619** [reply]
Hi! Could you please tell us which device you’re using?

**Reply by Tygopit** [reply]
Hi, 
I am considering using the SenseCAP Watcher in environments without access to an app or WiFi network—essentially as an intelligent photo trap.

I would like the ability to set a task, such as: **"If a person is detected, save the captured image directly to the SD card."** This way, I can periodically retrieve the device and review the images **on a separate device**, such as a computer or tablet, by simply accessing the SD card.

**Comment by wesleybasso** [comment]
Subject: SenseCAP Watcher - Identity/Provisioning Issue (Missing Device EUI & Key)
Context
I am a Mechatronics Engineering student working with the SenseCAP Watcher. My goal is to integrate it into a local Homelab environment and use the official AI features. However, I’ve encountered a critical "orphan device" issue where the hardware seems to lack its digital identity.

The Problem
The device is functional, boots correctly, and connects to the local network (IP assigned), but it is missing the core credentials required for the SenseCAP Mate App and Portal binding:

Device EUI: Returns empty or null.

Device Key: Missing/Not found.

Without these, I cannot bind the device to the official ecosystem, effectively locking me out of OTA updates and official AI services.

Technical Observations & Attempted Fixes
Serial Console (AT Commands): I accessed the device via Serial to check the provisioning data. The fields for EUI and Key appear to be uninitialized or corrupted in the factory partition.

Firmware Flashing: I attempted to re-flash/update the firmware, but without the original identity bound to the serial number, the official tools cannot "identify" the unit to authorize the link.

Local API: The device is reachable via IP, but since it's not bound to an account, many features remain restricted.

Request for Help
I am looking for insights from anyone who has dealt with ESP32-based SenseCAP provisioning:

Is there a specific AT Command or Python script to retrieve the EUI/Key directly from a protected flash partition if it wasn't wiped?

Has anyone successfully "re-provisioned" a Watcher using the Serial Number (SN) as a seed?

Are there alternative methods to bypass the "SenseCAP Mate" binding for local-only AI processing (e.g., custom firmware that doesn't rely on Seeed's cloud)?

If you have experience with the factory initialization process of these units or know a workaround to restore the identity, I would greatly appreciate your technical input.

Hardware Details:

Model: SenseCAP Watcher

Status: IP active, SN visible, EUI/Key null.
